net/arp/arp_send.c

```diff
@@ -372,7 +372,9 @@ int arp_send(in_addr_t ipaddr)
       /* Increment the retry count */
 
       state.snd_retries++;
-      nerr("ERROR: arp_wait failed: %d\n", ret);
+      nerr("ERROR: arp_wait failed: %d, ipaddr: %d.%d.%d.%d\n", ret,
+           (int)(ipaddr & 0xff), (int)((ipaddr >> 8) & 0xff),
+           (int)((ipaddr >> 16) & 0xff), (int)(ipaddr >> 24));
```

Comment:
```suggestion
      nerr("ERROR: arp_wait failed: %d, ipaddr: %d.%d.%d.%d\n", ret,
           ip4_addr1(ipaddr), ip4_addr2(ipaddr),
           ip4_addr3(ipaddr), ip4_addr4(ipaddr));
```

Reply:
Actually I'm not sure about this change. I just looked into `inet_ntoa_r` implementation and see that it does similar to current code. I will fill a separate issue for this.

Reply:
Yes. We need to use `ip4_addrX` macro in common code. `inet_ntoa_r` works as expected as is pronto byte by byte. Your code will not work well in BE systems

Reply:
@anchao could you please take a look?

Reply:
Here is the change https://github.com/apache/nuttx/pull/10301